core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -52,12 +55,22 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
    */
   @transient private var addedAbsPathFiles: mutable.Map[String, String] = null
 
+  @transient private var partitionPaths: mutable.Set[String] = null
+
+  @transient private var stagingDir: Path = _
```

Comment:
I mean, we can turn stagingDir into `private def stagingDir` or a private variable in a function.

Similarly, `partitionPaths` can be computed as `filesToMove.map(_.getPath).distinct` during the commit phase.